network-api/networkapi/utility/middleware.py

```diff
@@ -52,3 +54,19 @@ def __call__(self, request):
                 return HttpResponseTemporaryRedirect(redirect_url)
 
         return self.get_response(request)
+
+
+# Middleware to normalize the pt-br key errors in URLs by redirecting to pt-BR
+class NormalizeLocaleMiddleware(MiddlewareMixin):
+    def process_request(self, request):
+        # Split the path into segments
+        path_segments = request.path_info.strip("/").split("/")
+
+        # Check if the first segment is 'pt-br'
+        if len(path_segments) > 0 and path_segments[0] == "pt-br":
+            # Change only the country code to uppercase and reconstruct the path
+            path_segments[0] = "pt-BR"
+            new_path = "/" + "/".join(path_segments)
+            return redirect(new_path)
```

Comment:
```suggestion
            # Preserve query parameters
            if request.META.get("QUERY_STRING"):
                new_path += f"?{request.META['QUERY_STRING']}"

            return redirect(new_path)
```


As mentioned in my review, I noticed that query parameters (e.g., `?form=donate`) are being omitted during redirects. 

This change should do the trick. 👍

Reply:
Thanks for the great feedback @danielfmiranda! It seemed quick enough to generalize for both languages so I included that in the PR. Ready again for review!